geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentKafkaDataStore.scala

```diff
@@ -8,19 +8,28 @@
 
 package org.locationtech.geomesa.kafka.confluent
 
-import java.net.URL
-
 import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient
+import org.apache.avro.Schema
 import org.locationtech.geomesa.kafka.confluent.ConfluentGeoMessageSerializer.ConfluentGeoMessageSerializerFactory
 import org.locationtech.geomesa.kafka.data.KafkaDataStore
 import org.locationtech.geomesa.kafka.data.KafkaDataStore.KafkaDataStoreConfig
 import org.opengis.feature.simple.SimpleFeatureType
 
+import java.net.URL
+
 object ConfluentKafkaDataStore {
 
-  def apply(config: KafkaDataStoreConfig, schemaRegistryUrl: URL): KafkaDataStore = {
-    val metadata = new ConfluentMetadata(new CachedSchemaRegistryClient(schemaRegistryUrl.toExternalForm, 100))
-    val serialization = new ConfluentGeoMessageSerializerFactory(schemaRegistryUrl)
+  def apply(
+      config: KafkaDataStoreConfig,
+      schemaRegistryUrl: URL,
+      schemaOverrides: Map[String, (SimpleFeatureType, Schema)]): KafkaDataStore = {
+    val sftNameToSchema = schemaOverrides.values.map { case (sft, schema) => sft.getTypeName -> schema }.toMap
```

Comment:
this strikes me as slightly problematic since there's nothing enforcing unique feature type names, so they might clobber each other. maybe it's not really an issue in regular use

Reply:
@elahrvivaz The SFTs in the schemaOverrides map are generated using schemaToSft, where the name of the SFT is the topic in the config override. Thus, the SFTs cannot have overlapping names because if a topic appears more than once in the config only one will be used since the (topic -> config) are stored in a map. I could have called schemaToSft later to avoid passing a (SFT, Schema) (as the schema is directly used to create the SFT), but I wanted to validate the schema as early as possible. Also, because the schema overrides are passed in on creation of the DS, there could not be an SFT that already exists with the same name, and if you try to create another SFT with the same name it won't let you? Does that make sense / address the issue you mentioned?

Reply:
it makes sense, it would just be nice to enforce that with the api vs relying on the code creating the Map. maybe we could just change this line to be:
```suggestion
    val sftNameToSchema = schemaOverrides.map { case (topic, (_, schema)) => topic -> schema }
```

Reply:
Yeah that makes more sense.